### ISSUE TITLE
precompiled startup cache doesn't work

### ISSUE THREAD

**Issue by rmottola** [rmottola/Arctic-Fox#162]
On certain systems and configurations  precompiled startup cache fails to package.

Workaround is to add:
`ac_add_options --disable-precompiled-startupcache`

to the mozconfig.

Probably some file has a JS error or is missing, but I am unable to determine which one. The error is not the same on al systems and re-launching "mach package" shows a different file in the trace.

**Comment by accountboy** [comment]
Hey there! My son is in awe of the huuuuge iMac mom dragged home from work. We were excited to set up Arctic Fox, but hit a snag. 

Machine Details:
- Model: iMac (iMac7,1)
- Processor: Intel Core 2 Duo, 2.4 GHz
- Memory: 4 GB
- OS: OS X 10.6.8 (i386)

I dragged Arctic Fox into Applications, but when I try to launch, it crashes with an error related to `EXC_BREAKPOINT (SIGTRAP)` and a failure to load the `libSystem.B.dylib` library.

Haven't tried your workaround with the precompiled startup cache issue yet. Any pointers? I'll hold tight for now as compiling on our own seems a bit daunting. Thanks for all the work you put into this!

**Comment by rmottola** [comment]
@accountboy I don't think you are hitting  precompiled cache issue: that is a build problem currently, it can be disabled and the resulting binary will be slower to startup, but work nevertheless.
You might hit this issue: https://github.com/rmottola/Arctic-Fox/issues/135
So if you don't have macports installed (as developers have) you might miss certain libraries. I don't relink packages compared to what Matt used to do.
I provide libraries to be installed manually into e.g. /usr/lib . Please pe sure to use the maching version of the downloaded arcticfox release, 64bit vs 32bit. You mention a 64bit capable CPU, but then say "i386" (which would be 32bit)  at the os version.

**Comment by accountboy** [comment]
Hey, thank you for the quick response and clarifications. I apologize for any confusion regarding the 64-bit and 32-bit mix-up. I’ll take a look at issue #135 and try the manual installation of the libraries for the 32-bit version. Will get back to you if I have any further questions or run into issues. Thanks again for your help!

**Comment by rmottola** [comment]
> Hey, thank you for the quick response and clarifications. I apologize for any confusion regarding the 64-bit and 32-bit mix-up. I’ll take a look at issue #135 and try the manual installation of the libraries for the 32-bit version. Will get back to you if I have any further questions or run into issues. Thanks again for your help!

Your CPU and Mac are 64bit. So most probably you are running 64bit. Please check opening "Activity Monitor" and seek in the column Kind (add it if needed) if there are some "Intel (64bit)" processes running.
Try using Arctic Fox 64bit and its associated libraries. I stress that this issue is not associated with the precompiled cache. Maybe open a discussion for support or specific issue.

**Comment by accountboy** [comment]
Wizardry! I can confirm all processes are running at 64 bit per above. Can’t browse github on the browsers of the era so need to check back with usb stick transfer of the 64 bit Arctic Fox. Thank you.

Edit:

Confirmed working! I installed library files in the /usr/local/lib/ if that extra download and step wasn't obvious.

**Comment by rmottola** [comment]
@roytam1 do you have a clue about this? I suppose one of the JS files has an issue, maybe a preprocessor issue or an include issue. I am unable however to understand which file !

**Comment by rmottola** [comment]
> Hey, thank you for the quick response and clarifications. I apologize for any confusion regarding the 64-bit and 32-bit mix-up. I’ll take a look at issue #135 and try the manual installation of the libraries for the 32-bit version. Will get back to you if I have any further questions or run into issues. Thanks again for your help!

that's really great news and what this project is about: try to give good old hardware still some usefulness.

**Comment by roytam1** [comment]
> @roytam1 do you have a clue about this?

can't tell since it is not happening here

**Comment by rmottola** [comment]
> > @roytam1 do you have a clue about this?
> 
> can't tell since it is not happening here

no need ti disable it on windows? "mach package" works fine?
Interesting, it is broken both on linux and mac. So it may be some configuration I have (e.g. webrtc or other things enabled) or something is not used or is specific to windows. Would be intersting and a clue by itself.

**Comment by roytam1** [comment]
> "mach package" works fine?

yeah, without doing anything.

**Comment by rmottola** [comment]
> > "mach package" works fine?
> 
> yeah, without doing anything.

interesting.... I have it both on Linux and Mac currently.
This could hint into some preprocessor difference, so either in a specific file or some ifdef somwhere.
I am unable to understand where the issue comes from! If I could get at least the file or the directory, I could diff that against FF or TFF.

**Comment by rmottola** [comment]
Things improved a bit and on FreeBSD and NetBSD with WebRTC disabled precompile cache can be left enabled. However, on Linux disabling WebRTC doesn't help! strange!

**Comment by rmottola** [comment]
Now it needs to be disabled again on FreeBSD... this is a mystery! If someone can help me debugging the file that fails, I can't understand from the logs.

**Comment by rmottola** [comment]
It works again on most platforms
